packages/frontend/src/components/post/NestedReply.tsx

```diff
@@ -15,22 +15,73 @@ interface IReplyProps extends IPostWithReplies {
 export const resolveNestedReplyThreads = (
   allPosts: IPostWithReplies[],
   depth: number,
+  postsVisibilityMap: Record<string, number> | undefined,
+  setPostsVisibility: Dispatch<SetStateAction<Record<string, number> | undefined>>,
   onSuccess: () => void,
+  trail: string[],
+  additionalDataKeys: string[][],
+  setAdditionalDataKeys: Dispatch<SetStateAction<string[][]>>,
+  shouldRerenderThreads: MutableRefObject<boolean>,
   writerToShow?: string,
 ) => {
   const replyNodes: React.ReactNode[] = [];
-  const proof = '';
+
+  if (
+    !postsVisibilityMap ||
+    !allPosts ||
+    (allPosts.length > 0 && !postsVisibilityMap[allPosts[0].id])
+  ) {
+    return (
+      <div>
+        <button
+          onClick={() => {
+            if (!allPosts) {
+              const newKeys = [...additionalDataKeys];
+              newKeys.push(trail);
+              shouldRerenderThreads.current = true;
+              setAdditionalDataKeys(newKeys);
+            } else {
+              const newPostsVisibility = { ...postsVisibilityMap };
+              allPosts.forEach((post) => {
+                newPostsVisibility[post.id] = 1;
+              });
+              setPostsVisibility(newPostsVisibility);
+            }
+          }}
+        >
+          {allPosts ? allPosts.length : 'View'} more replies
```

Comment:
Just a small copy thing here: 
```suggestion
          {allPosts ? allPosts.length : 'View'} more {allPosts?.length === 1 ? 'reply' : 'replies'}
```